src/FilePreviewer.js

```diff
@@ -26,206 +51,208 @@ import ViewportContent from './ViewportContent';
-        }
-
-        setFile(file);
-        setTotalPages(1);
-        setCurrentPage(0);
-    }, [props.file]);
-
-    return (
-        <div ref={viewportRef} onClick={props.onClick} className="preview-wrapper">
-            <PreviewBar
-                onPageUp={onPageUp}
-                onRotate={onRotate}
-                totalPages={totalPages}
-                onDownload={onDownload}
-                onPageDown={onPageDown}
-                currentPage={currentPage}
-                hidden={props.hideControls}
-            />
-
-            <ViewportContent
-                file={file}
-                containerRef={containerRef}
-                onPageChange={setCurrentPage}
-                onLoadSuccess={onLoadSuccess}
-            />
-
-            <ViewportControl
-                onZoomIn={onZoomIn}
-                onZoomOut={onZoomOut}
-                onFitToScreen={onFitToScreen}
-                hidden={props.hideControls}
-            />
-        </div>
+    // Scroll the viewport to the page position.
+    containerRef.current.scrollTop = previousPage.offsetTop - 10;
+  };
+
+  /**
+   * Handler. Scroll to the next page and update the index.
+   *
+   * @return {void}
+   */
+  const onPageDown = () => {
+    const nextIndex = R.clamp(0, totalPages, currentPage + 1);
+
+    setCurrentPage(nextIndex);
+
+    const nextPage = document.querySelector(`div[data-pdfpage="${nextIndex}"]`);
+
+    // Scroll the viewport to the page position.
+    containerRef.current.scrollTop = nextPage.offsetTop - 10;
+  };
+
+  /**
+   * Handler. Scale all the pages or image
+   * up by a factor of 0.25.
+   *
+   * @return {void}
+   */
+  const onZoomIn = () => {
+    setUsingFitToScreen(false);
+    setFile(setZoomIn(file));
+  };
+
+  /**
+   * Handler. Scale all the pages or image
+   * down by a factor of 0.25.
+   *
+   * @return {void}
+   */
+  const onZoomOut = () => {
+    setUsingFitToScreen(false);
+    setFile(setZoomOut(file));
+  };
+
+  /**
+   * Handler. Rotate the PDF (all pages) or Image.
+   *
+   * @return {void}
+   */
+  const onRotate = () => {
+    setFile(setNewRotation(file));
+  };
+
+  /**
+   * Handler. Fit to screen.
+   *
+   * (This activates the "using fit to screen" mode, which
+   * makes the document or image resize based on the viewport size.)
+   *
+   * @return {void}
+   */
+  const onFitToScreen = () => {
+    setUsingFitToScreen(true);
+
+    // Get the "fit to screen" scale.
+    const newScale = originalSizes.map(originalSize =>
+      getFitToScreenScale(viewportSize, originalSize),
     );
+
+    setFile(prevValue => R.assoc('scale', newScale, prevValue));
+  };
+
+  /**
+   * Handler. Download the current file, it can be a
+   * PDF or Image, as `url` or `base64` content.
+   *
+   * @return {void}
+   */
+  const onDownload = () => {
+    const url = file.url || `data:${file.mimeType};base64,${file.data}`;
+    return saveFile(url, file.name || 'download.pdf');
```

Comment:
Add a method for `getFilename` that uses file.name if provided or defaults to `download.{ext}`

Reply:
Done!